bigfastapi/customer.py

```diff
@@ -0,0 +1,125 @@
+from datetime import datetime
+from typing import List
+from fastapi import APIRouter
+from bigfastapi.models import organisation_models
+from bigfastapi.utils.utils import generate_short_id
+from bigfastapi.schemas import customer_schemas
+from bigfastapi.models import customer_models
+import sqlalchemy.orm as orm
+import fastapi
+from bigfastapi.db.database import get_db
+from uuid import uuid4
+from fastapi.responses import JSONResponse
+from fastapi import status
+
+app = APIRouter(tags=["Customers 💁"],)
+
+
+
+
+@app.post("/customers", response_model=customer_schemas.CustomerCreateResponse)
+def create_customer(customer: customer_schemas.CustomerValidator, db: orm.Session = fastapi.Depends(get_db)):
+    organization = db.query(organisation_models.Organization).filter(organisation_models.Organization.name == customer.organization_name).first()
+    if organization is not None:
+        customer_instance = customer_models.Customer(
+            id = uuid4().hex,
+            customer_id = generate_short_id(size=12),
+            first_name = customer.first_name,
+            last_name= customer.last_name,
+            organization= organization.name,
+            email= customer.email,
+            phone_number= customer.phone_number,
+            address= customer.address,
+            gender= customer.gender,
+            age= customer.age,
+            postal_code= customer.postal_code,
+            language= customer.language,
+            country= customer.country,
+            city= customer.city,
+            region= customer.region,
+            date_created = datetime.now(),
+            last_updated = datetime.now()
+
+        )
+        db.add(customer_instance)
+        db.commit()
+        db.refresh(customer_instance)
+        return {"message": "Customer created succesfully", "customer": customer_schemas.CustomerInDB.from_orm(customer_instance)}
+    else:
+        return {"message": "Organization Doesn't Exist", "customer": {}}
+
+@app.get('/customers', response_model=List[customer_schemas.CustomerInDB])
+def get_customers(db: orm.Session = fastapi.Depends(get_db)):
+    customers = db.query(customer_models.Customer).all()
```

Comment:
What if there are 10 million customers? You return them all?

Reply:
Added pagination